Project-Uni/RootShare
# Request To Speak Feature

1. Viewers can now request host to become a guest speaker
2. Event supports up to 2 simultaneous guest speakers
3. Redesigned Manage Speakers Modal to include request to speak feature and incorporate useModal and useForm
4. Noticed that stream continues even if host refreshes their webpage, so updated UI and functionality to handle that ("Start Live Stream" button defaults to "End Live Stream" if stream is still going)

### Testing
 - Access event as host and at least two viewers and invite viewers through searchbar and "Request To Speak" button
 (you can open two distinct incognito windows by opening one from personal chrome account and one from RootShare Dev chrome account)
 - Try to refresh the page as host and make sure functionality continues as expected
 - Refresh viewers/guest speakers to make sure nothing breaks (There is still some weirdness that I couldn't get rid of due to OpenTok intricacies, but nothing game-breaking)
 - End Live Stream with guest speakers and make sure speakers get removed cleanly
